app/src/main/java/com/dscvit/handly/util/RecyclerViewExtensions.kt

```diff
@@ -0,0 +1,39 @@
+package com.dscvit.handly.util
+
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+
+interface OnItemClickListener {
+    fun onItemClicked(position: Int, view: View)
+}
+
+fun RecyclerView.addOnItemClickListener(onClickListener: OnItemClickListener) {
+    this.addOnChildAttachStateChangeListener(object: RecyclerView.OnChildAttachStateChangeListener {
+        override fun onChildViewDetachedFromWindow(view: View) {
+            view.setOnClickListener(null)
+        }
+
+        override fun onChildViewAttachedToWindow(view: View) {
+            view.setOnClickListener{
+                val holder = getChildViewHolder(view)
+                onClickListener.onItemClicked(holder.adapterPosition, view)
+            }
+        }
+    })
+}
+
+fun RecyclerView.addOnItemLongClickListener(onClickListener: OnItemClickListener) {
+    this.addOnChildAttachStateChangeListener(object: RecyclerView.OnChildAttachStateChangeListener {
+        override fun onChildViewDetachedFromWindow(view: View) {
+            view.setOnClickListener(null)
+        }
+
+        override fun onChildViewAttachedToWindow(view: View) {
+            view.setOnLongClickListener{
+                val holder = getChildViewHolder(view)
+                onClickListener.onItemClicked(holder.adapterPosition, view)
+                return@setOnLongClickListener true
```

Comment:
Funfact: You can skip the `return` in a lambda, and it will return the result of the last statement in the block. In this case, removing return will still return `true`.

Reply:
Ooo interesting